src/app/src/components/Submissions/Detail/Map.js

```diff
@@ -19,40 +30,103 @@ export default function Map({ submission }) {
             >
                 <MapPanes>
                     <DefaultBasemap />
-                    <MapButtons submission={submission} />
+                    <MapButtons
+                        id={id}
+                        status={status}
+                        email={submission.primary_contact.email}
+                        shape={submission.shape}
+                    />
+                    <Polygon shape={submission.shape} />
                 </MapPanes>
             </MapContainer>
-            <SubmissionStatusBar submission={submission} />
+            <SubmissionStatusBar status={status} />
         </>
     );
 }
 
-function MapButtons({ submission }) {
+function MapButtons({ id, status, email, shape }) {
+    const navigate = useNavigate();
+    const userRole = useSelector(state => state.auth.user.role);
+
+    const canEditDraft =
+        userRole === ROLES.CONTRIBUTOR || userRole === ROLES.ADMINISTRATOR;
+    const canReview =
+        userRole === ROLES.VALIDATOR || userRole === ROLES.ADMINISTRATOR;
+
+    const getDrawButtonText = () => {
+        switch (status) {
+            case BOUNDARY_STATUS.DRAFT:
+                return canEditDraft ? 'Edit boundary' : null;
+            case BOUNDARY_STATUS.SUBMITTED:
+                return canReview ? 'Start review' : null;
```

Comment:
Clicking "Start review" should create a row in the Review table

Reply:
339721f